core/src/main/java/org/bitcoinj/coinjoin/CoinJoin.java

```diff
@@ -185,12 +190,14 @@ public static boolean isCollateralValid(Transaction txCollateral, boolean checkI
         }
 
         // the collateral tx must not have been seen on the network
-        boolean hasBeenSeen = txCollateral.getConfidence().getConfidenceType() == TransactionConfidence.ConfidenceType.UNKNOWN;
+        TransactionConfidence confidence = txCollateral.getConfidence();
+        boolean hasBeenSeen = confidence.getConfidenceType() != TransactionConfidence.ConfidenceType.UNKNOWN ||
+                confidence.numBroadcastPeers() > 0 || confidence.getIXType() != TransactionConfidence.IXType.IX_NONE;
```

Comment:
Previously there was a bug here that would flag every collateral tx (used for coinjoin fees) as already spent regardless.  Now we will check 3 things to determine if it was spent yet.

These transactions differ in that they are created by DashJ, but not committed to the wallet and not sent to the network using the normal means of sending transactions.  Rather they are sent to masternodes and will be sent to the network by the mixing masternode when certain conditions are met.  Most collateral tx's are never spent.